cartography/intel/azure/key_vaults.py

```diff
@@ -21,65 +19,93 @@
 def load_key_vaults(session: neo4j.Session, subscription_id: str, data_list: List[Dict], update_tag: int) -> None:
     session.write_transaction(_load_key_vaults_tx, subscription_id, data_list, update_tag)
 
+
 def load_key_vaults_keys(session: neo4j.Session, subscription_id: str, data_list: List[Dict], update_tag: int) -> None:
     session.write_transaction(_load_key_vaults_keys_tx, subscription_id, data_list, update_tag)
 
+
 def load_key_vaults_secrets(session: neo4j.Session, subscription_id: str, data_list: List[Dict], update_tag: int) -> None:
     session.write_transaction(_load_key_vault_secrets_tx, subscription_id, data_list, update_tag)
 
+
 def load_key_vaults_certificates(session: neo4j.Session, subscription_id: str, data_list: List[Dict], update_tag: int) -> None:
     session.write_transaction(_load_key_vault_certificates_tx, subscription_id, data_list, update_tag)
 
-@timeit
-def get_key_vault_keys_client(credentials: Credentials, vault_url: str) -> KeyClient:
-    client = KeyClient(vault_url, credentials.arm_credentials)
-    return client
-
-@timeit
-def get_key_vault_secrets_client(credentials: Credentials, vault_url: str) -> SecretClient:
-    client = SecretClient(vault_url,credentials)
-    return client
 
 @timeit
 def get_key_vault_certificates_client(credentials: Credentials, vault_url: str) -> CertificateClient:
     client = CertificateClient(vault_url, credentials)
     return client
 
+
 @timeit
 def get_key_vaults_client(credentials: Credentials, subscription_id: str) -> KeyVaultManagementClient:
     client = KeyVaultManagementClient(credentials, subscription_id)
     return client
 
 
 @timeit
-def get_key_vaults_list(client: KeyVaultManagementClient, regions: List, common_job_parameters: Dict) -> List[Dict]:
+def get_key_vaults_list(client: KeyVaultManagementClient) -> List[Dict]:
     try:
         key_vaults_list = list(map(lambda x: x.as_dict(), client.vaults.list_by_subscription()))
         return key_vaults_list
     except HttpResponseError as e:
         logger.warning(f"Error while retrieving key vaults - {e}")
         return []
 
+
 @timeit
-def transform_key_vaults(key_vaults: List[Dict], regions: List, common_job_parameters: Dict) -> List[Dict]:
+def transform_key_vaults(client: KeyVaultManagementClient, key_vaults: List[Dict], regions: List, common_job_parameters: Dict) -> List[Dict]:
     key_vaults_data = []
     for vault in key_vaults:
         x = vault['id'].split('/')
         vault['resource_group'] = x[x.index('resourceGroups') + 1]
         vault['consolelink'] = azure_console_link.get_console_link(
             id=vault['id'], primary_ad_domain_name=common_job_parameters['Azure_Primary_AD_Domain_Name'])
+        List_Certificate_Permissions = False
+        for policy in vault.get('properties', {}).get('access_policies', []):
```

Comment:
we should move this code out to a different function and call that function from here. So that it's modular.